front/src/features/books/booksSlice.tsx

```diff
@@ -0,0 +1,81 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import api from "api/index";
+import { initialState } from "./helperData";
+
+import { IfetchBooksData, IAddItemToBasket } from "../interfaces/index";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: initialState,
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.status = "inProgress";
+    },
+    fetchBooksDataSucceeded: (
+      state,
+      action: PayloadAction<IfetchBooksData>
+    ) => {
+      state.data = [...action.payload.data];
+      state.status = "succeeded";
+    },
+    fetchBooksDataFailed: (state) => {
+      state.status = "failed";
+    },
+    addItemToBasket: (state, action: PayloadAction<IAddItemToBasket>) => {
+      const { id } = action.payload;
+      const isInBasket = state.basket.find((item) => item.id === id);
+
+      const item = { ...action.payload, value: 1 };
+
+      state.basket = isInBasket
+        ? state.basket.map((basketItem) =>
+            basketItem.id === id
+              ? {
+                  ...basketItem,
+                  value: basketItem.value + 1,
+                }
+              : basketItem
+          )
+        : [...state.basket, item];
+    },
+  },
+});
+
+export const {
+  fetchBooksDataStarted,
+  fetchBooksDataSucceeded,
+  fetchBooksDataFailed,
+  addItemToBasket,
+} = booksSlice.actions;
+
+export const fetchBooks = (pageNumber: number) => async (dispatch: any) => {
+  dispatch(fetchBooksDataStarted());
+
+  try {
+    const data = await api.books.fetchBooks(pageNumber);
+    dispatch(fetchBooksDataSucceeded(data.data));
+  } catch (error) {
+    dispatch(fetchBooksDataFailed());
+  }
+};
+export const selectBooksData = (state: any) => state.books.data;
+export const selectBooksBasket = (state: any) => state.books.basket;
+
+interface Item {
```

Comment:
"Item" jest za ogólne, nazwij to np. BasketItem

Reply:
teraz widzę, że to zdublowany IBook - zaimportuj IBook i tutaj też użyj.
Takie rzeczy dobrze trzymać w folderze interfaces (albo models, poczytaj co Ci bardziej pasuje, interfaces wola bardziej angularowcy :P) np. src/interfaces/books.ts

Reply:
Done.